File: Semester 5/Mobile application programming/Labs/Lab05/bank_accounts_server/lib/account.dart
class Account {
  late var id;
  var iban;
  var sold;
  var currency;
  var bank;
  var name;

  Account({
    this.id,
    this.iban,
    this.sold,
    this.currency,
    this.bank,
    this.name
  });

  // Account(this.iban, this.sold, this.currency, this.bank, this.name) {
  //   id = -1;
  // }

  factory Account.fromMap(Map<String, dynamic> json) => new Account(
        id: json['id'],
        iban: json['iban'],
        sold: json['sold'],
        currency: json['currency'],
        bank: json['bank'],
        name: json['name'],
      );

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'iban': iban,
      'sold': sold,
      'currency': currency,
      'bank': bank,
      'name': name
    };
  }
}

File: Semester 5/Mobile application programming/Labs/Lab05/bank_accounts_backend/BankAccounts/src/main/java/com/example/bankaccounts/entity/Account.java
package com.example.bankaccounts.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;

import java.util.Objects;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@ToString
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String iban;
    private String sold;
    private String currency;
    private String bank;
    private String name;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Account account = (Account) o;
        return Objects.equals(id, account.id) && Objects.equals(iban, account.iban) && Objects.equals(sold, account.sold) && Objects.equals(currency, account.currency) && Objects.equals(bank, account.bank) && Objects.equals(name, account.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, iban, sold, currency, bank, name);
    }
}
